src/utils/network.rs

```diff
@@ -0,0 +1,12 @@
+use std::{net::TcpListener};
+
+pub(crate) fn find_port(default_port: u16) -> Option<u16> {
```

Comment:
define a second parameter to chose how many ports should be tried?

Reply:
Not needed IMO, as said in the other comment :)